packages/interface/src/components/ImageUpload.tsx

```diff
@@ -1,82 +1,94 @@
 import { useMutation } from "@tanstack/react-query";
 import clsx from "clsx";
 import { ImageIcon } from "lucide-react";
-import { type ComponentProps, useRef, useCallback } from "react";
+import { type ComponentProps, ElementType, forwardRef, useRef, useCallback } from "react";
 import { Controller, useFormContext } from "react-hook-form";
 import { toast } from "sonner";
 
 import { IconButton } from "~/components/ui/Button";
 
+import { PolymorphicRef } from "./ui";
+
 export interface IImageUploadProps extends ComponentProps<"img"> {
   name?: string;
   maxSize?: number;
 }
 
-export const ImageUpload = ({
-  name = "",
-  maxSize = 1024 * 1024, // 1 MB
-  className,
-}: IImageUploadProps): JSX.Element => {
-  const ref = useRef<HTMLInputElement>(null);
-  const { control } = useFormContext();
-
-  const select = useMutation({
-    mutationFn: async (file: File) => {
-      if (file.size >= maxSize) {
-        toast.error("Image too large", {
-          description: `The image to selected is: ${(file.size / 1024).toFixed(2)} / ${(maxSize / 1024).toFixed(2)} kb`,
-        });
-        throw new Error("IMAGE_TOO_LARGE");
-      }
+export const ImageUpload = forwardRef(
```

Comment:
curious why change the `ImageUpload` here which should not be related to the search?
I saw you mentioned 2 errors here:
> A not-found file in the div style
> A ref error involving the Form component and the ImageUpload component

if this error is not because of your changes, then maybe fix it in another PR?

Reply:
I added a couple of housekeeping fixes 😊 (that is why I am modifying the `ImageUpload` component). I also updated the PR description with some of the things I worked on